app/lib/actions/katello/content_view_version/create_export_history.rb

```diff
@@ -0,0 +1,67 @@
+module Actions
+  module Katello
+    module ContentViewVersion
+      class CreateExportHistory < Actions::EntryAction
+        input_format do
+          param :smart_proxy_id, Integer
+          param :exporter_data, Hash
+          param :content_view_version_id, Integer
+          param :from_content_view_version_id, Integer
+          param :destination_server, String
+        end
+
+        output_format do
+          param :export_history_id, Integer
+          param :path, String
+        end
+
+        def run
+          smart_proxy = ::SmartProxy.find(input[:smart_proxy_id])
+          api = ::Katello::Pulp3::Api::Core.new(smart_proxy)
+          Rails.logger.info input[:exporter_data][:pulp_href]
+          export_data = api.export_api.list(input[:exporter_data][:pulp_href]).results.first
+          output[:exported_file_checksum] = export_data.output_file_info
+          file_name = output[:exported_file_checksum].first&.first
+          path = File.dirname(file_name.to_s)
+          output[:export_path] = path
+          cvv = ::Katello::ContentViewVersion.find(input[:content_view_version_id])
+          from_cvv = ::Katello::ContentViewVersion.find(input[:from_content_view_version_id]) unless input[:from_content_view_version_id].blank?
+
+          export_metadata = ::Katello::Pulp3::ContentViewVersion::Export.new(
+                                                     content_view_version: cvv,
+                                                     smart_proxy: smart_proxy,
+                                                     from_content_view_version: from_cvv).generate_metadata
+
+          toc_path_info = output[:exported_file_checksum].find { |item| item.first.end_with?("toc.json") }
+          export_metadata[:toc] = File.basename(toc_path_info.first)
+
+          history = ::Katello::ContentViewVersionExportHistory.new(
+            content_view_version_id: input[:content_view_version_id],
+            destination_server: input[:destination_server],
+            path: path,
+            metadata: export_metadata
+          )
+          history.export_type = history.export_type_from_metadata
+          history.audit_comment = generate_audit_comment(history, User.current, cvv)
+          history.save!
```

Comment:
It's not normally recommended to do DB actions in the `run` method.  But

1. it's how it was before;
2. it's only a single record so shouldn't be too risky
3. I need access to all of the above data inside `input`, which is not populated within `plan`.  for example, if I try to use `input[:exporter_data][:pulp_href]` from inside `plan`, I get a value like `"Step(3)[:exporter_data][:pulp_href]"` (as a string) instead of the populated value.